services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala

```diff
@@ -183,7 +183,69 @@ class GitLabProjectVersioningTest extends AsyncWordSpec with ScalaFutures with M
           .getFile(project, path, Some(version))
           .map(_ shouldBe Left(VersioningException("Exception. Response status: 404")))
       }
+    }
 
+    "getFileCommits" should {
+      val dummyCommitJson: String = s"[${Json.stringify(Json.toJson(dummyFileCommit))}]"
+      val path: Path = Paths.get("tmp/foo.txt")
+      def request(project: Project) =
+        mockHttpClient.get(
+          url = gitLabConfig.url + "projects/" + project.repository.get.value + "/repository/files/" + path.toString,
+          headers = gitLabConfig.token
+        )
+      "return list of Project versions with 200 response" taggedAs Service in {
+        when(request(withRepoProject))
+          .thenReturn(Future.successful(Response(HttpStatusCodes.OK, dummyCommitJson, Map())))
+        gitLabProjectVersioning.getFileCommits(withRepoProject, path).map {
+          _ shouldBe Right(Seq(dummyFileCommit))
+        }
+      }
+      "throw new VersioningException with 400 response" taggedAs Service in {
+        when(request(withRepoProject))
+          .thenReturn(Future.successful(Response(HttpStatusCodes.BadRequest, EmptyBody, Map())))
+        gitLabProjectVersioning.getFileCommits(withRepoProject, path).map {
+          _ shouldBe Left(VersioningException("Could not take the file commits. Response status: 400"))
+        }
+      }
+    }
+
+    "getFileVersions" should {
+      val dummyCommitJson: String =
+        s"[${Json.stringify(Json.toJson(dummyFileCommit))}, ${Json.stringify(Json.toJson(dummyFileCommitAux))}]"
```

Comment:
dummyFileCommitAux is confusing. I think you just should create another dummyFileCommit like so: TestProjectUtils.getDummyFileCommit()
Also use Json.stringify(Json.toJson(List(dummyFileCommit, anotherDummyFileCommit))) instead, it's easy to read.

Reply:
I join this opinion

Reply:
I changed to use _Json.toJson(List_. Also, I renamed the variable dummyFileCommitAux to dummyExisitngFileCommit, i cannot create a new one cause the process needs a commit that matches with a version